src/test/java/com/google/sps/AbuseDetectionTest.java

```diff
@@ -0,0 +1,225 @@
+package com.google.sps;
+
+import com.google.sps.algorithms.AbuseDetection;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalTime;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+/** */
+@RunWith(JUnit4.class)
+public final class AbuseDetectionTest {  
+
+  public void add10Requests(AbuseDetection instance){
```

Comment:
For clarity, maybe add the approximate time of the added requests to the method name (here and below).

Reply:
Did that